File: 3rd Semester/Advanced Programming Methods/exams session/trains/java/repository/Repository.java
package ubb.scs.map.examen.repository;

import ubb.scs.map.examen.domain.Entity;

import java.util.Optional;


public interface Repository<ID, E extends Entity<ID>>{
        Optional<E> findOne(ID id);
        Iterable<E> findAll();
}

File: 4th Semester/Design and Programming Environments/labs/1st lab/C#/repository/Repository.cs
using basker.domain;

namespace basker.repository;

public interface Repository<ID, E> where E : Entity<ID>
{
    void Save(E entity);
    void Update(E entity);
    void Delete(ID id);
    E findOne(ID id);
    IEnumerable<E> findAll();
}
